src/core/DependencyInjection.ts

```diff
@@ -0,0 +1,109 @@
+import { Context } from 'aws-lambda';
+
+import DependencyAwareClass from './DependencyAwareClass';
+import { LambdaWrapperConfig } from './config';
+
+// eslint-disable-next-line no-use-before-define
+type Class<T> = new (di: DependencyInjection) => T;
+
+/**
+ * Dependency injection container.
+ *
+ * Dependencies (singleton instances of dependency-aware classes) are provided
+ * to the main Lambda handler and other dependencies via this class.
+ */
+export default class DependencyInjection {
+  /**
+   * Instantiated dependencies.
+   */
+  readonly dependencies: Record<string, DependencyAwareClass>;
+
+  /**
+   * True until all dependencies have been constructed.
+   */
+  private isConstructing = true;
+
+  constructor(
+    readonly config: LambdaWrapperConfig,
+    readonly event: any,
+    readonly context: Context,
+  ) {
+    this.dependencies = Object.fromEntries(
+      Object.entries(config.dependencies)
+        .map(([, Constructor]) => [Constructor.name, new Constructor(this)]),
```

Comment:
I think I mentioned this one prior to your leave, and I can't remember the outcome of the conversation, so just flagging it here again as a reminder:

could we use `Object.values` instead?

Reply:
Yes, good spot. This is left over from an earlier iteration where I was using the keys for something, but now they are ignored.

Reply:
https://github.com/comicrelief/lambda-wrapper/pull/882/commits/a3d12b381cedd77ed879e6b044c1f3afd35d7314